native/core/src/execution/planner.rs

```diff
@@ -1155,12 +1154,9 @@ impl PhysicalPlanner {
                     ))
                 });
 
-                let object_store = object_store::local::LocalFileSystem::new();
-                // register the object store with the runtime environment
-                let url = Url::try_from("file://").unwrap();
-                self.session_ctx
-                    .runtime_env()
-                    .register_object_store(&url, Arc::new(object_store));
+                // By default, local FS object store registered
+                // if `hdfs` feature enabled then HDFS file object store registered
+                let object_store_url = register_object_store(Arc::clone(&self.session_ctx))?;
```

Comment:
Should we update this function ([get_file_path](https://github.com/apache/datafusion-comet/blob/996362e78d497c02542f1e29dbb7cba3ec16f64c/native/core/src/parquet/util/jni.rs#L214C8-L214C21)) as well? 
It's currently used by `NATIVE_ICEBERG_COMPAT` but the goal is to unify it with `COMET_DATAFUSION`.

Reply:
Thats a good point, to verify it we probably need to read Iceberg from HDFS which can be done in #1367

Reply:
We don't need to wait for actual iceberg integration. CometScan will use COMPAT_ICEBERG if the configuration is set (That's how we are able to run the unit tests).

Reply:
@comphead we can log a follow up issue to update `get_file_path` if you like.

Reply:
Thanks @parthchandra lets create a followup ticket. Appreciate if you do it as I'm afraid I can miss some Iceberg details in ticket description

Reply:
https://github.com/apache/datafusion-comet/issues/1407. There's no detail in the PR. Can you assign to me if possible, and I'll remember to take care of it.